src/main/java/com/pusher/rest/data/TriggerResult.java

```diff
@@ -0,0 +1,58 @@
+package com.pusher.rest.data;
+
+import java.util.Map;
+
+import com.google.gson.Gson;
+
+public class TriggerResult {
+    static class Data {
+        public final Map<String, String> event_ids;
+
+        public Data(final Map<String, String> event_ids) {
+            this.event_ids = event_ids;
+        }
+    }
+
+    private static final Gson GSON = new Gson();
+
+    private final Result.Status status;
+    private final Integer httpStatus;
+    private final String message;
+
+    public TriggerResult(final Result result) {
+        this.status = result.getStatus();
+        this.httpStatus = result.getHttpStatus();
+        this.message = result.getMessage();
+    }
+
+    /**
+     * Get the enum classifying the result of the call
+     */
+    public Result.Status getStatus() {
+        return status;
+    }
+
+    /**
+     * Get the HTTP status code of the call, useful for debugging instances of UNKNOWN_ERROR
+     */
+    public Integer getHttpStatus() {
+        return httpStatus;
+    }
+
+    /**
+     * Get the data response (success) or descriptive message (error) returned from the call
+     */
+    public String getMessage() {
+        return message;
```

Comment:
> I would like to keep API compatibility with the Result (should I introduce an IResult ?) 

Depends what you want to achieve. Source compatibility, you already have.

But it you want people to be able to handle collections of different `*Result`s then you need an interface specifying which methods are common between them. Generic error handling code might be an example.

If you do introduce one:
 - Consider if subclassing might actually be the cleanest way. After all TriggerResult IS A Result does hold true.
 - Please don't call it `IResult`. What is interface and what is concrete is an implementation detail not of concern to the end user of the library. It should be called `Result` and the existing concrete one should move to be `SimpleResult` or similar. Although, as I mention above, it could also stay where it is an serve as the base class for specialisations.